libs/payments/ui/src/lib/client/components/CouponForm/index.tsx

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use client';
+
+import { Localized } from '@fluent/react';
+import { SubmitButton } from 'libs/payments/ui/src/lib/client/components/SubmitButton';
+import { ButtonVariant } from 'libs/payments/ui/src/lib/server/components/BaseButton';
+import { updateCartAction } from '../../../actions/updateCart';
+import { getFallbackTextByFluentId } from '../../../utils/error-ftl-messages';
+
+interface WithCouponProps {
+  cartId: string;
+  cartVersion: number;
+  couponCode: string;
+  readOnly: boolean;
+}
+
+export const WithCoupon = ({
+  cartId,
+  cartVersion,
+  couponCode,
+  readOnly,
+}: WithCouponProps) => {
+  async function removeCoupon() {
+    await updateCartAction(cartId, cartVersion, { couponCode: '' });
+  }
+
+  return (
+    <>
+      <h2 className="m-0 mb-4 font-semibold text-grey-600">
+        <Localized id="next-coupon-promo-code-applied">
+          Promo Code Applied
+        </Localized>
+      </h2>
+
+      <form
+        action={removeCoupon}
+        className="flex gap-4 justify-between items-center"
+        data-testid="coupon-hascoupon"
+      >
+        <span className="break-all">{couponCode}</span>
+        {readOnly ? null : (
+          <span>
+            <SubmitButton
+              variant={ButtonVariant.Secondary}
+              data-testid="coupon-remove-button"
+            >
+              <Localized id="next-coupon-remove">Remove</Localized>
+            </SubmitButton>
+          </span>
+        )}
+      </form>
+    </>
+  );
+};
+
+interface WithoutCouponProps {
+  cartId: string;
+  cartVersion: number;
+  readOnly: boolean;
+}
+
+export const WithoutCoupon = ({
+  cartId,
+  cartVersion,
+  readOnly,
+}: WithoutCouponProps) => {
+  async function applyCoupon(formData: FormData) {
+    const promotionCode = formData.get('coupon') as string;
+
+    await updateCartAction(cartId, cartVersion, {
+      couponCode: promotionCode,
+    });
+  }
+
+  const error = null;
+
+  return (
+    <>
+      <h2 className="m-0 mb-4 font-semibold text-grey-600">
+        <Localized id="next-coupon-promo-code">Promo Code</Localized>
+      </h2>
+
+      <form action={applyCoupon} data-testid="coupon-form">
```

Comment:
One of the goals for the SP3 frontend was to use a component library where possible.

Even though it's not necessary, since everything works as expected in this case, I'm still leaning towards using the radix Forms component, so that it's consistent throughout the App.

However, this could also be done as a polish PR. What do you think about using Radix Forms here, and doing it as a polish PR?